### ISSUE TITLE
Host header check

### ISSUE THREAD

**Issue by plynchnlm** [http-party/http-server#758]
#### What's the problem this feature will solve?

Defend against DNS rebinding attacks (https://en.wikipedia.org/wiki/DNS_rebinding). 

#### Describe the solution you'd like

Add a new option for specifying the permitted value of the Host HTTP header (i.e. the hostname of the server to which the
request is sent).

A key part of the DNS rebinding attack is that the user's browser will be sending a request to an internal server with an invalid Host header for that server, i.e., the Host header's value is not a real DNS name for the IP address on which the server is running.  If the server checks the Host header against the value provided in the new option, the attack fails.  For an example of a server that does this, see https://angular.io/cli/serve and its "--host" option.

#### Alternative Solutions

I don't see any other solution other than to use a different package.

#### Additional context

Note that DNS rebinding is an issue even when running the server on a development machine behind a firewall.  Any one who has access to the develop machine server who also browses to a malicious website can open up access to that development server.

**Comment by plynchnlm** [comment]
I corresponds to ng serve's --host parameter.  If a server did not have a hostname, then you would not be able to use --host to specify it.

**Comment by velzie** [comment]
`ng serve`'s --host is already implemented as `-a` for http-server. I assume you meant --allowed-hosts?

**Comment by plynchnlm** [comment]
No, --allowed-hosts is about the hosts that can connect to the server.  This is stretching my memory a bit, but I think -a in http-server controls the IP address being listened on, while --host in ng serve controls which HTTP Host header value the server will accept (i.e., the host name of the server).   In a DNS rebinding attack, the malicious request would arrive with some other Host header value, and be rejected.

**Comment by velzie** [comment]
Seems to be `--allowed-hosts` that implements the host header functionality you mention. Passing a server hostname into `--host` just throws an error

<img width="1122" height="318" alt="Image" src="https://github.com/user-attachments/assets/c874db83-35cf-4f21-884f-fccbb237cc89" />

<img width="1155" height="366" alt="Image" src="https://github.com/user-attachments/assets/3d3f0909-2165-4d3b-9ddf-03f5253d71ac" />

<img width="929" height="289" alt="Image" src="https://github.com/user-attachments/assets/cf48a396-3141-4fd6-b1e4-3d5e15c9475e" />

**Comment by plynchnlm** [comment]
Okay.  "--allowed-hosts" for ng serve is documented as "List of hosts that are allowed to access the dev server".  That sounds like it is a list of the remote (client) hosts connecting to the dev server, but the behavior in your screenshots shows that it really is checking the Host headers, which is what is needed.   The pull request https://github.com/http-party/http-server/pull/953 implements the needed behavior, and I agree with closing this ticket, though I still find the comment "list of hosts allowed to access the server" confusing.  Thanks for adding this support.
